pkg/frontend/v2/frontend.go

```diff
@@ -228,7 +229,12 @@ enqueueAgain:
 		return nil, ctx.Err()
 
 	case resp := <-freq.response:
-		return resp, nil
+		if stats.ShouldTrackHTTPGRPCResponse(resp.HttpResponse) {
+			stats := stats.FromContext(ctx)
+			stats.Merge(resp.Stats)
```

Comment:
I'd add comment that it's ok if `stats` is nil here, otherwise it looks like it may panic.